src/main/java/org/apache/ibatis/migration/options/OptionsParser.java

```diff
@@ -53,41 +53,7 @@ private static boolean parseOptions(String arg, SelectedOptions options) {
       final String[] argParts = arg.substring(2).split("=");
       final Options option = Options.valueOf(argParts[0].toUpperCase());
 
-      switch (option) {
-        case PATH:
-          options.getPaths().setBasePath(new File(argParts[1]));
-          break;
-        case ENVPATH:
-          options.getPaths().setEnvPath(new File(argParts[1]));
-          break;
-        case SCRIPTPATH:
-          options.getPaths().setScriptPath(new File(argParts[1]));
-          break;
-        case DRIVERPATH:
-          options.getPaths().setDriverPath(new File(argParts[1]));
-          break;
-        case HOOKPATH:
-          options.getPaths().setHookPath(new File(argParts[1]));
-          break;
-        case ENV:
-          options.setEnvironment(argParts[1]);
-          break;
-        case FORCE:
-          options.setForce(true);
-          break;
-        case TRACE:
-          options.setTrace(true);
-          break;
-        case HELP:
-          options.setHelp(true);
-          break;
-        case TEMPLATE:
-          options.setTemplate(argParts[1]);
-          break;
-        case IDPATTERN:
-          options.setIdPattern(argParts[1]);
-          break;
-      }
+      option.accept(options, argParts.length > 1 ? argParts[1] : null);
```

Comment:
Probably my fault but the logic is flawed in that it looks at = `--foo=true`  with a shortcut of `-f true` breaks this.

Reply:
How is this any different then what was there before? All of these, if they use the argument use `argParts[1]` if anything the original code was flawed and would allow for an `ArrayIndexOutOfBoundsException` if you had an arg like `--foo=`

It looks like, if anything the regex could be bit more robust: `"-([a-z])|--([a-z]+)[ =](.*)"` then use a `Pattern` matcher instead of a split and extract the groups to match. 

Better yet would be to put that logic in the `Option` class with a factory method like: `Option.of(arg)` so you could have something like:

`Set<Option> selected = Arrays.stream(args).map(Option::of).collect(toSet());`

Reply:
How is this any different then what was there before?

It's not, I'm pointing out a logic flaw that I missed a while back.
The regex fix is a good idea. (although your [ =] breaks on multiple spaces without a trim).
I don't want to get cute or complicated w a fix here. There's already plans to investigate options parsing library which better handles these. Although, to be fair we don't have a ticket so you wouldn't have known that.

Reply:
To be honest: Spring boot/spring cli does a fantastic job of this.

Reply:
`-([a-z])|--([a-z]+)[ ]+?=[ ]+?(.*)]`
how's that?

any number of spaces before or after the "=" actually, now we are just talking about properties and you're just better off doing something simple like this and passing it to properties:

```java
args = Arrays.stream(args).filter( s -> s.startsWith("-")).map(a -> a.subString(1));
args = Arrays.stream(args).filter( s -> s.startsWith("-")).map(a -> a.subString(1));
Arrays.stream(args).map(a -> String.format("%s%n",a))
.reduce(String::concat).map(String::toByteArray)
.map(ByteArrayInputStream::new).map(bais -> {
    Properties p = new Properties();
    p.loadFrom(bais);
   return p;
}.ifPresent(props -> ... apply options);
```
I wanted to get smarter with a lambda recursion but you get the point.

Parsing properties is trickier than it seems - hence the reason to simply package it up and hand it off to `Properties` -- don't I wish that class would take an `InputStream` constructor